emu_base/pulser_adapter.py

```diff
@@ -169,10 +183,8 @@ def _extract_omega_delta_phi(
                     locals_a_d_p[q_id]["phase"][t1] + locals_a_d_p[q_id]["phase"][t2]
                 ) / 2.0
             # omegas at different times need to have the laser waist applied independently
-            if t1 in global_times:
-                omega_1 *= waist_factors
-            if t2 in global_times:
-                omega_2 *= waist_factors
+            omega_1 *= global_times_to_amp_factors.get(t1, 1.0)
```

Comment:
Nothing related to this change, but could you move those two lines outside of the `q_pos, q_id` loop, it's weird now cuz they are defined in the loop and used outside
```
t1 = math.floor(t)
t2 = math.ceil(t)
```

Reply:
done